docs/DeveloperGuide.md

```diff
@@ -2,37 +2,243 @@
+* It will then call the corresponding `list()` method defined in `RecipeList`. 
+* This `list()` method in turn calls the `ListRecipeCommand.execute()` command. 
+* The reason for such a structure is to allow for command validation *before* the real execution. A good example is the `find()` command. 
+* Thus, `RecipeList` contains an intermediate method for each functionality (add, find, delete, so on) , and serves as a command validator.
+  * **Note:** these methods have been omitted in the class diagram for brevity.
+* In `add()` and `delete()`, the `saveRecipes()` method is called to save the recipeBook after the modification.
+
+
+
+---
+
+## Appendix: Requirements
 
 ## Product scope
+
 ### Target user profile
 
-{Describe the target user profile}
+* Has a need to manage a significant number of recipes.
+  * For example, professional culinary practitioners, students who love to cook.
+* Can type fast 
+* People having special dietary restrictions.
+* People who would like easily access the recipes they found online.
 
 ### Value proposition
 
-{Describe the value proposition: what problem does it solve?}
+* The user will be able to add, access, and list recipes.
+* He can also filter recipes based on ingredients, time required, dietary restrictions, etc.
+* The saved recipes will be loaded in a recipe text file for easy sharing with friends.
+* The user can put the recipes into different categories.
 
 ## User Stories
 
-|Version| As a ... | I want to ... | So that I can ...|
-|--------|----------|---------------|------------------|
-|v1.0|new user|see usage instructions|refer to them when I forget how to use the application|
-|v2.0|user|find a to-do item by name|locate a to-do without having to go through the entire list|
+| Version | As a ... | I want to ...                              | So that I can ...                                             |
+|---------|----------|--------------------------------------------|---------------------------------------------------------------|
+| v1.0    | new user | see usage instructions                     | refer to them when I forget how to use the application        |
+| v1.0    | user     | add a recipe                               | curate my own unique recipes for future use                   |
+| v1.0    | user     | delete a recipe                            | remove any outdated or unimportant recipes                    |
+| v2.0    | user     | find a recipe by name                      | locate a recipe without having to go through the entire list  |
+| v2.0    | foodie   | save and load recipes                      | share recipes with friends conveniently                       |
+| v2.0    | user     | find recipes without a certain ingredient  | save time finding recipes that I can eat                      |
+| v2.0    | user     | find a recipe by date                      | find a recipe if I don't remember its name                    |
```

Comment:
I like that you list the user stories to be implemented for each version, will give new developers a better idea of what to start working on and what they can expect.